src/main/components/Browse/Browse.js

```diff
@@ -30,15 +30,120 @@ class Browse extends Component {
 
   constructor(props) {
     super(props);
-    this.handleAnalysesChange = this.handleAnalysesChange.bind(this);
+
+    this.clearFilters = this.clearFilters.bind(this);
     this.handleAnalysisClick = this.handleAnalysisClick.bind(this);
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.handleOptions = this.handleOptions.bind(this);
     this.state = {
-      isOpen: true,
-      analyses: null
+      analyses: null,
+      chosenFilters: {
+        title: null,
+        description: null,
+        jiraId: null,
+        libraryIds: null,
+        sampleIds: null,
+        project: null
+      },
+      isOpen: true
     };
   }
 
-  handleAnalysesChange = analyses => this.setState({ analyses: analyses });
+  isFilterNull = filter => !filter;
+
+  isFilterLibraryOrSample = filter => Array.isArray(filter);
+
+  isFilterInArray = (analysisFilter, chosenFilter) =>
+    analysisFilter.includes(chosenFilter.label);
+
+  doFiltersMatch = (analysisFilter, chosenFilter) =>
+    JSON.stringify(analysisFilter) === JSON.stringify(chosenFilter.label);
+
+  /* 
+  Is the filter null? (ie. not present in the current selection)   
+    If so, then return true to the chain of ands, having no effect on the overall expression
+    If not, is the information being filtered an array in the analysis (ie. libraryIds or sampleIds)?
+      If so, does that array contain the chosen libraryId or sampleId?
+        If so, return true
+        If not, return false  
+      If not, then check if the filter in the selection matches the data in the analysis in question
+        If so, return true
+        If not, return false
+  */
+  isFilterWithinAnalysis = (analysis, filter, filters) =>
+    this.isFilterNull(filters[filter]) ||
+    (this.isFilterLibraryOrSample(analysis[filter])
+      ? this.isFilterInArray(analysis[filter], filters[filter])
+      : this.doFiltersMatch(analysis[filter], filters[filter]));
+
+  isWithinFilter = (analysis, filters) =>
+    /*
+    reduce() has an initialValue of true
+    if a false is added, the bool return will evaluate to false
+    otherwise, it will evaluate to true
+    */
+    Object.keys(filters).reduce(
+      (result, filter) =>
+        result && this.isFilterWithinAnalysis(analysis, filter, filters),
+      true
+    );
+
+  handleFilterChange = (dashboardAnalyses, name) => value => {
+    this.setState(
+      {
+        chosenFilters: {
+          ...this.state.chosenFilters,
+          [name]: value
+        }
+      },
+      () => {
+        const filters = this.state.chosenFilters;
+        let analyses = dashboardAnalyses;
+        analyses = analyses.filter(analysis =>
+          this.isWithinFilter(analysis, filters)
+        );
+        this.setState({ analyses: analyses });
+      }
+    );
+  };
+
+  handleOptions = (currentAnalyses, filter) =>
```

Comment:
This function is more appropriate in Filters, as its sole purpose is to take the analysis data and format in a way to pass along to react-select.